migration/identity-migration/src/test/java/io/camunda/migration/identity/GroupMigrationHandlerTest.java

```diff
@@ -140,6 +140,53 @@ void setErrorWhenGroupCreationHasError() {
                       .noneMatch(MigrationStatusUpdateRequest::success);
                 }));
     verify(managementIdentityClient, times(2)).fetchGroups(anyInt());
-    verify(groupService, times(2)).createGroup(any());
+    verify(groupService, times(2)).createGroup(any(GroupDTO.class));
+  }
+
+  @Test
+  public void shouldNormalizeGroupIDIfTooLong() {
+    // given
+    final String longGroupName = "a".repeat(300);
+    final Group group = new Group("id1", longGroupName);
+    when(managementIdentityClient.fetchGroups(anyInt()))
+        .thenReturn(List.of(group))
+        .thenReturn(List.of());
+    when(groupService.createGroup(any(GroupDTO.class)))
+        .thenReturn(CompletableFuture.completedFuture(null));
+
+    // when
+    migrationHandler.migrate();
+
+    // then
+    final var groupResult = ArgumentCaptor.forClass(GroupDTO.class);
+    verify(groupService, times(1)).createGroup(groupResult.capture());
+    assertThat(groupResult.getValue().groupId())
+        .describedAs("Group ID should be normalized to 256 characters")
+        .hasSize(256);
+    assertThat(groupResult.getValue().groupId())
+        .describedAs("Group ID should be lowercased")
+        .isEqualTo(longGroupName.toLowerCase().substring(0, 256));
+  }
+
+  @Test
+  public void shouldNormalizeGroupIDIfContainsUnsupportedCharacters() {
+    // given
+    final String groupNameWithUnsupportedChars = "Group@Name#With$Special%Chars";
+    final Group group = new Group("id1", groupNameWithUnsupportedChars);
+    when(managementIdentityClient.fetchGroups(anyInt()))
+        .thenReturn(List.of(group))
+        .thenReturn(List.of());
+    when(groupService.createGroup(any(GroupDTO.class)))
+        .thenReturn(CompletableFuture.completedFuture(null));
+
+    // when
+    migrationHandler.migrate();
+
+    // then
+    final var groupResult = ArgumentCaptor.forClass(GroupDTO.class);
+    verify(groupService).createGroup(groupResult.capture());
+    assertThat(groupResult.getValue().groupId())
+        .describedAs("Group ID should be normalized to remove unsupported characters")
+        .doesNotContain("#", "$", "%");
```

Comment:
🔧  just to check that the expected underscore is indeed present
```suggestion
        .describedAs("Group ID should be normalized to replace unsupported characters with `_`")
        .doesNotContain("#", "$", "%")
        .contains("_");
```